src/ProjectOrigin.VerifiableEventStore/Services/Repository/ITransactionRepository.cs

```diff
@@ -0,0 +1,21 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using ProjectOrigin.VerifiableEventStore.Models;
+
+namespace ProjectOrigin.VerifiableEventStore.Services.Repository;
+
+public interface ITransactionRepository
+{
+    Task Store(StreamTransaction @event);
+
+    Task<ImmutableLog.V1.Block?> GetBlockFromTransactionHash(TransactionHash transactionHash);
+    Task<TransactionStatus> GetTransactionStatus(TransactionHash transactionHash);
```

Comment:
Should this be 

```suggestion
    Task<TransactionStatus?> GetTransactionStatus(TransactionHash transactionHash);
```

Reply:
Actually no, the TransactionStatus is the following enum

```csharp
public enum TransactionStatus
{
    Unknown = 0,
    Pending = 1,
    Failed = 2,
    Committed = 3
}
```

so it simply returns Unknown = 0